### PR TITLE
Cscfairmeta 459 add subject heading field qvain light

### PR DESCRIPTION
**Pull request by kasperkeinanen**
Add the subject heading field to Qvain-Light + create "AddedValue" component to be reused in all multivalue field components and improved its accessibility.

### CODE REVIEW

File: etsin_finder/frontend/js/stores/view/qvain.js [hunk 1]
```diff
@@ -803,6 +841,14 @@ class Qvain {
     // Keywords
     this.keywordsArray = researchDataset.keyword || []
 
+    // SubjectHeadings/Themes
```
Comment (tahme) [inline etsin_finder/frontend/js/stores/view/qvain.js:844]: clear the subjectHeadingsArray here to make sure it has no leftovers from other datasets

**Comment by tahme** [comment]
There's a bug when a subject heading is missing a translation. I think we've used the `und` translation in those cases for other fields. For example, try to have Qvain in English and pick the Finnish subject heading "albumiinimenetelmä". This produces an error:

> The prop `text` is marked as required in `AddedValue`, but its value is `undefined`.

I'm not sure about the language selector. Our other reference data fields don't have one (even though the reference data has translations) so it's a bit weird that one of the fields has a language selector. Maybe it would be better to have the query match any language like other similar fields do.

**Comment by kasperkeinanen** [comment]
"There's a bug when a subject heading is missing a translation" good catch @tahme , This is a bug I forgot to fix 👍  

About the language selector, the subject headings field is quite different from the other reference data because of the huge amount of data that is queried. This also means that if we don't specify in what language we are querying the reference data, it will give very different results. For example, without specifying the language if you query "home" it would return results from all languages related to "home", and depending on the language it will have a totally different meaning. This often leads to poor usability like in the example, if the user wanted to have words related to mold (finish "home") and instead gets results related to houses. This is the reason why I decided to put the language selector. With the language selector, the user can decide in what language he/she wants to query the reference data to be able to add the subject headings most relevant to the dataset.

If you want to try the difference when querying all languages, replace line 31 in `getReferenceData.js` with 
```js
`${METAX_FAIRDATA_ROOT_URL}/es/reference_data/keyword/_search?size=100&filter_path=hits.hits._source&q=label.\\*:${inputValue}*`
```

I am open to discussion related to this, but this is my view of the best way to let the user add the most relevant subject headings in an intuitive way.

**Review by tahme — APPROVED** [review]

**Review by tahme — CHANGES_REQUESTED** [review]
Sorry for canceling approval, thought I checked that the bug I mentioned was fixed but it wasn't yet.

**Comment by tahme** [comment]
There's still a bug related to missing translations.
I can now add `albumiinimenetelmä` but when I save the dataset, it disappears from the list.

I think the bug is related to this initialization code in `qvain.js` where `pref_label.en` (might be undefined) is used explicitly.
```javascript
    this.subjectHeadingsArray = []
    if (researchDataset.theme !== undefined) {
      researchDataset.theme.forEach(element => {
        this.subjectHeadingValue = SubjectHeading(element.pref_label.en, element.identifier, element.pref_label)
        this.addToSubjectHeadingsArray()
      })
    }
```

**Comment by MattiasLevlin** [comment]
Replaced by https://github.com/CSCfi/etsin-finder/pull/619

Closing this PR.

**Comment by MattiasLevlin** [comment]
The replacement PR has been merged. Deleting this one.

**Pull request closed without merging**